src/components/global/PlaygroundTabs/index.tsx

```diff
@@ -0,0 +1,231 @@
+    if (isValidElement(child) && isTabItem(child)) {
+      return child;
+    }
+    // child.type.name will give non-sensical values in prod because of
+    // minification, but we assume it won't throw in prod.
+    throw new Error(
+      `Docusaurus error: Bad <Tabs> child <${
+        // @ts-expect-error: guarding against unexpected cases
+        typeof child.type === 'string' ? child.type : child.type.name
+      }>: all children of the <Tabs> component should be <TabItem>, and every <TabItem> should have a unique "value" prop.`
+    );
+  });
+  const values =
+    valuesProp ??
+    // Only pick keys that we recognize. MDX would inject some keys by default
+    children.map(({ props: { value, label, icon, attributes } }) => ({
+      value,
+      label,
+      attributes,
+      icon,
+    }));
+  const dup = duplicates(values, (a, b) => a.value === b.value);
+  if (dup.length > 0) {
+    throw new Error(
+      `Docusaurus error: Duplicate values "${dup
+        .map((a) => a.value)
+        .join(', ')}" found in <Tabs>. Every value needs to be unique.`
+    );
+  }
+  // When defaultValueProp is null, don't show a default tab
+  const defaultValue =
+    defaultValueProp === null
+      ? defaultValueProp
+      : defaultValueProp ?? children.find((child) => child.props.default)?.props.value ?? children[0]?.props.value;
+  if (defaultValue !== null && !values.some((a) => a.value === defaultValue)) {
+    throw new Error(
+      `Docusaurus error: The <Tabs> has a defaultValue "${defaultValue}" but none of its children has the corresponding value. Available values are: ${values
+        .map((a) => a.value)
+        .join(', ')}. If you intend to show no default tab, use defaultValue={null} instead.`
+    );
+  }
+
+  const [selectedValue, setSelectedValue] = useState(defaultValue);
+  const tabRefs: (HTMLLIElement | null)[] = [];
+  const { blockElementScrollPositionUntilNextRender } = useScrollPositionBlocker();
+
+  const handleTabChange = (event: React.FocusEvent<HTMLLIElement> | React.MouseEvent<HTMLLIElement>) => {
+    const newTab = event.currentTarget;
+    const newTabIndex = tabRefs.indexOf(newTab);
+    const newTabValue = values[newTabIndex]!.value;
+
+    if (newTabValue !== selectedValue) {
+      blockElementScrollPositionUntilNextRender(newTab);
+      setSelectedValue(newTabValue);
+    }
+  };
+
+  const handleKeydown = (event: React.KeyboardEvent<HTMLLIElement>) => {
+    let focusElement: HTMLLIElement | null = null;
+
+    switch (event.key) {
+      case 'ArrowRight': {
+        const nextTab = tabRefs.indexOf(event.currentTarget) + 1;
+        focusElement = tabRefs[nextTab] || tabRefs[0]!;
+        break;
+      }
+      case 'ArrowLeft': {
+        const prevTab = tabRefs.indexOf(event.currentTarget) - 1;
+        focusElement = tabRefs[prevTab] || tabRefs[tabRefs.length - 1]!;
+        break;
+      }
+      default:
+        break;
+    }
+
+    focusElement?.focus();
+  };
+
+  useEffect(() => {
```

Comment:
Why is this needed?

Reply:
Updated the base `button` example, since we are going to delete that before merging `component-playground` into `main`. You will be able to see the arrow navigation behavior in the live demo.

This code is responsible for hiding the arrow navigation initially, until there is horizontal scroll. There is a slight jump for the left nav button when it is initially rendered (once reaching the 40px offset), but discussed with Ben and this was ok for the first iteration. The goal is to not display the arrow navigation until they are needed.